Crashes if TypeScript compilation fails during Karma launch

**Setup:**
Angular CLI: 1.6.1
Karma: 1.7.1

**Steps to Reproduce:**
1) Supply an invalid number of parameters to a method in TypeScript code, TypeScript compilaiton will fail with error TS2554: Expected X arguments, but got Y.
2) Karma launch will fail
3) As a result of this, karma-tfs-reporter crashes since the Karma spec does not have start or finish timestamps.

```
TypeError: Invalid date
    at TypeError (native)
    at C:\Development\Test UI\Main\My.Test\node_modules\dateformat\lib\dateformat.js:40:17
    at toISOString (C:\Development\Test UI\Main\My.Test\node_modules\karma-tfs-reporter\trx.js:14:12)
    at module.exports (C:\Development\Test UI\Main\My.Test\node_modules\karma-tfs-reporter\trx.js:119:24)
    at TfsReporter.onExit (C:\Development\Test UI\Main\My.Test\node_modules\karma-tfs-reporter\index.js:53:44)
    at Server.<anonymous> (C:\Development\Test UI\Main\My.Test\node_modules\karma\lib\events.js:13:22)
    at emitOne (events.js:101:20)
    at Server.emit (events.js:188:7)
    at Server.EventEmitter.emitAsync (C:\Development\Test UI\Main\My.Test\node_modules\karma\lib\events.js:72:10)
    at disconnectBrowsers (C:\Development\Test UI\Main\My.Test\node_modules\karma\lib\server.js:383:10)
    at process.<anonymous> (C:\Development\Test UI\Main\My.Test\node_modules\karma\lib\server.js:405:5)
    at emitOne (events.js:96:13)
    at process.emit (events.js:188:7)
    at process.emit (C:\Development\Test UI\Main\My.Test\node_modules\ts-node\node_modules\source-map-support\source-map-support.js:439:21)
    at process._fatalException (bootstrap_node.js:296:26)
```

--- a/trx.js
+++ b/trx.js
@@ -31,11 +31,11 @@ function escape(str) {
 };
 
 module.exports = function (testResults) {
-    var start = finish = null;
+    var start = Math.min.apply(testResults.specs.map(spec => spec.start.getTime()));
+    var finish = Math.max.apply(testResults.specs.map(spec => spec.finish.getTime()));
 
-    if(testResults.specs.length) {
-        start = Math.min.apply(testResults.specs.map(spec => spec.start.getTime()));
-        finish = Math.max.apply(testResults.specs.map(spec => spec.finish.getTime()));
+    if(!testResults.specs.length) {
+        start = finish = new Date();
     }
 
     var specs = testResults.specs.map(spec => {

--- a/trx.js
+++ b/trx.js
@@ -31,8 +31,12 @@ function escape(str) {
 };
 
 module.exports = function (testResults) {
-    var start = Math.min.apply(null, testResults.specs.map(spec => spec.start.getTime()));
-    var finish = Math.max.apply(null, testResults.specs.map(spec => spec.finish.getTime()));
+    var start = finish = null;
+
+    if(testResults.specs.length) {
+        start = Math.min.apply(testResults.specs.map(spec => spec.start.getTime()));
+        finish = Math.max.apply(testResults.specs.map(spec => spec.finish.getTime()));
+    }
 
     var specs = testResults.specs.map(spec => {
         spec.suite = escape(spec.suite);

--- a/trx.js
+++ b/trx.js
@@ -31,8 +31,8 @@ function escape(str) {
 };
 
 module.exports = function (testResults) {
-    var start = Math.min.apply(testResults.specs.map(spec => spec.start.getTime()));
-    var finish = Math.max.apply(testResults.specs.map(spec => spec.finish.getTime()));
+    var start = Math.min.apply(null, testResults.specs.map(spec => spec.start.getTime()));
+    var finish = Math.max.apply(null, testResults.specs.map(spec => spec.finish.getTime()));
 
     if(!testResults.specs.length) {
         start = finish = new Date();

